unitTests/ui/PagerTest.ts

```diff
@@ -79,17 +77,32 @@ export function PagerTest() {
       expect(test.env.queryStateModel.set).toHaveBeenCalledWith('first', (currentPage - 1) * 10);
     });
 
-    it('should update page when state is changed', () => {
-      test = Mock.advancedComponentSetup<Pager>(
-        Pager,
-        new Mock.AdvancedComponentSetupOptions(undefined, undefined, env => {
-          return env.withLiveQueryStateModel();
-        })
-      );
-      test.cmp.setPage(7);
-      expect(test.cmp.currentPage).toBe(7);
-      test.env.queryStateModel.set('first', 30);
-      expect(test.cmp.currentPage).toBe(4);
+    describe('when query state model is updated', () => {
+      beforeEach(() => {
+        test = Mock.advancedComponentSetup<Pager>(
+          Pager,
+          new Mock.AdvancedComponentSetupOptions(undefined, undefined, env => {
+            return env.withLiveQueryStateModel();
+          })
+        );
+      });
+
+      it('should update current page when first result is changed', () => {
+        test.cmp.setPage(7);
+        expect(test.cmp.currentPage).toBe(7);
+        test.env.queryStateModel.set('first', 30);
```

Comment:
I would remove this line and the next from this test since we are changing the page using a different mechanism, and this mechanism is covered by the test on line 97.

Then in the test descriptions, we could specify `... is changed using the component method setPage`, and `...is changed by updating the QSM` to distinguish the approach used by each test. When one fails, we have a better idea of which kind of change is no longer working.